From b7940496b4f8efdccb9b4097b496b0d9b2af1eea Mon Sep 17 00:00:00 2001
From: Vojtech Trefny <vtrefny@redhat.com>
Date: Tue, 18 Jun 2024 14:47:39 +0200
Subject: [PATCH] tests: Try waiting after partition creation for XFS resize
 test

The test randomly fails to find the newly created partition so
lets try waiting a bit with udev settle.
---
 tests/storage_tests/formats_test/fs_test.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/storage_tests/formats_test/fs_test.py b/tests/storage_tests/formats_test/fs_test.py
index f3c9fef5a..5da4a9339 100644
--- a/tests/storage_tests/formats_test/fs_test.py
+++ b/tests/storage_tests/formats_test/fs_test.py
@@ -11,6 +11,7 @@
 from blivet.devices import PartitionDevice, DiskDevice
 from blivet.flags import flags
 from blivet.util import capture_output
+from blivet import udev
 
 from .loopbackedtestcase import LoopBackedTestCase
 
@@ -149,6 +150,7 @@ def _create_partition(self, disk, size):
         pend = pstart + int(Size(size) / disk.format.parted_device.sectorSize)
         disk.format.add_partition(pstart, pend, parted.PARTITION_NORMAL)
         disk.format.parted_disk.commit()
+        udev.settle()
         part = disk.format.parted_disk.getPartitionBySector(pstart)
 
         device = PartitionDevice(os.path.basename(part.path))
